Documentation content additions and revisions for M2 (#207)
datajoint/datajoint-docs: contents/definition/04-Definition-Syntax.rst

Invalid attribute names
   ``firstName``, ``first name``, ``2photon_scan``, ``two-photon_scan``, ``TwoPhotonScan``

> Attribute names should be unique across all tables.

Reviewer: ```suggestion
Ideally, attribute names should be unique across all tables that are likely to be used in queries together.
```

For example, tables often have attributes representing the start times of sessions, recordings, etc.
Such attributes must be uniquely named in each table, such as ``session_start_time`` or ``recording_start_time``.

Default values
--------------

Secondary attributes can be given default values.
A default value will be used for an attribute if no other value is given at the time the entity is :ref:`inserted <insert>` into the table.
Generally, default values are numerical values or character strings.
Default values for dates must be given as strings as well, contained within quotes (with the exception of ``CURRENT_TIMESTAMP``).
Note that default values can only be used when inserting as a mapping.
Primary key attributes cannot have default values (with the exceptions of ``auto_increment`` and ``CURRENT_TIMESTAMP`` attributes; see :ref:`primary-key`).

An attribute with a default value of ``NULL`` is called a **nullable attribute**.

> Nullable attributes can be thought of as optional, such as an attribute that may not have a value for every entity.

Reviewer: ```suggestion
A nullable attribute may be thought of as potentially inapplicable to a subset of entities, such as the attribute may not have a value at all.
```

Author: I don't find your suggested change very clear. Also I believe `can` is more appropriate here; `may` usually refers to permission rather than ability.

Reviewer: The key distinction is between "optional" and "inapplicable".   If an attribute is optional, then it's okay to keep it in the same table.  If the attribute is not applicable to a distinct subset of entities, they should be separated into a their own table.

Reviewer: For example, if students and employees are in the same table, then the attribute salary will only apply to employees but not students.  The attribute e-mail may be optional but applicable to all.

Author: Did you mean to post these comments under this discussion: https://github.com/datajoint/datajoint-docs/pull/207/files#r242705176 ?

The section here is about appropriately nullable attributes.

Reviewer: ```suggestion
A nullable attribute may be thought of as having an optional value that may be missing yet applicable to all entities in the table.  
If an attribute is inapplicable to some entities in a table, it probably means that the schema should be redesigned to split entities into separate tables where all attributes apply to all entities.
```

Author: ```suggestion
 A nullable attribute can be thought of as applying to all entities in a table but having an optional *value* that may be absent in some entities.
 Nullable attributes should *not* be used to indicate that an attribute is inapplicable to some entities in a table (see :ref:`normalization`).
```

Author: How's about that?

> However, nullable attributes should be used deliberately where their inclusion in an entity is meaningful; they should not be used as a shortcut to easy data entry.

Reviewer: ```suggestion
Nullable attributes should be used sparingly to indicate optional rather than inapplicable attributes that still apply to all entities in the table.
```
